app/src/main/java/com/arctouch/codechallenge/data/remote/MoviesRemoteSource.kt

```diff
@@ -0,0 +1,59 @@
+package com.arctouch.codechallenge.data.remote
+
+import android.annotation.SuppressLint
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.data.MoviesDataSource
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.android.schedulers.AndroidSchedulers
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+object MoviesRemoteSource: MoviesDataSource {
+    private var api: TmdbApi = Retrofit.Builder()
+                .baseUrl(TmdbApi.URL)
+                .client(OkHttpClient.Builder().build())
+                .addConverterFactory(MoshiConverterFactory.create())
+                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+                .build()
+                .create(TmdbApi::class.java)
+
+    @SuppressLint("CheckResult")
+    override fun getMovies(callback: MoviesDataSource.LoadMoviesCallback, page: Long) {
+        api.upcomingMovies(TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE, page)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe( {
+                    val moviesWithGenres = it.results.map { movie ->
+                        movie.copy(genres = Cache.genres.filter { g -> movie.genreIds?.contains(g.id) == true })
+                    }
+                    Cache.cacheMovies(moviesWithGenres)
+                    callback.onMoviesLoaded(Cache.movies)
+            },
+                { callback.onDataNotAvailable() })
+    }
+
+    override fun getMovie(movieId: Long, callback: MoviesDataSource.GetMovieCallback) {
+        api.movie(movieId, TmdbApi.API_KEY, TmdbApi.DEFAULT_LANGUAGE)
```

Comment:
A good practice way to avoid this warning it's using ```CompositeDisposable```. This way you are also helping with memory leak of the application :)

Reply:
Actually I have not used RxJava's `Observables` recently, so I was a bit confused with it. I usually use Retrofit's `Call`, but I didn't want to change it all because it's used in your base code.
However I've read more about it in the last few days after the challenge, and indeed it could be improved.